File: src/main/kotlin/a/SoldierAndBananas.kt
package a

/* Problem Link: https://codeforces.com/problemset/problem/546/A */

fun main() {
    val (k, n, w) = readln().split(" ").map { it.toInt() }
    var dollars = 0

    for (i in 1..w) dollars += i * k
    print(if (dollars <= n) 0 else dollars - n)
}

File: Codeforces/C++/problemset/Soldier and Bananas.cpp
// Problem Link: https://codeforces.com/problemset/problem/546/A

#include <iostream>
using namespace std;
int main()
{
    long long k,n,w,sum = 0;
    cin >> k >> n >> w;
    for(int i = 1; i <= w; i++)
        sum += i*k;
    if(n >= sum)
        cout << 0;
    else
        cout << sum - n;
    return 0;
}
